### PR TITLE
Release Notes 3/17

### PR DESCRIPTION
**Pull request by alvinli222**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes is cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* AKS patch versions 1.27.9 and 1.28.5 are now available with 1.27.9 used as the default version for new clusters.
+
+
+### Release notes
+* Features
+  * [5,000 Node Limit by Default](https://azure.microsoft.com/en-us/updates/generally-available-azure-kubernetes-service-aks-support-for-5k-node-limit-by-default-for-standard-tier-clusters/) is generally available in AKS. This limit is available for Standard tier and Premium tier clusters.
```
Comment (kaarthis) [inline CHANGELOG.md:15]: 1.29 should be GA with this release. If you dont find the PR i ll tag you for that in the RP channel @Alvin.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+
+
+* Preview features
+  * [Deployment Safeguards](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: 'Warning' and 'Enforcement'
+  * [Windows GPU Support](https://learn.microsoft.com/en-us/azure/aks/use-windows-gpu) on AKS is now available in preview.
+  * [Trusted Launch Support](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) on AKS is now available in preview.
+
+* Behavioral change
```
Comment (kaarthis) [inline CHANGELOG.md:23]: Are there no bug fixes or Behavioral features at all ? Its puzzling did you check all the PRs in the last 2-3 weeks that are marked customer facing ?

**Review by kaarthis — CHANGES_REQUESTED** [review]
PTAL - please check the PRs using the wiki for release notes process.

File: CHANGELOG.md (hunk 1 above)
Comment (alvinli222) [reply CHANGELOG.md:16]: added.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+
+* Bug fixes
+  * AgentPool PUT requests with node init taints will no longer fail if the request does not change the field.
+  * Fixed a bug with legacy clusters that blocked all new operations on these clusters.
+  * Fixed a bug where node taints may be overwritten on certain PUT requests.
+  * Fixed a bug where clusters running LTS could get a list of non-LTS versions to upgrade to.
+  * Fixed a bug with Application Gateway Ingress Controller where it is unable to fetch secret objects during cluster upgrade.
+
+
+* Component updates
```
Comment (wangyira) [inline CHANGELOG.md:38]: any vhd updates?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,48 @@
...
+  * Kubernetes 1.29 is [GA](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli).
+  * [5,000 Node Limit by Default](https://azure.microsoft.com/en-us/updates/generally-available-azure-kubernetes-service-aks-support-for-5k-node-limit-by-default-for-standard-tier-clusters/) is generally available in AKS. This limit is available for Standard tier and Premium tier clusters.
+
+* Preview features
+  * [Deployment Safeguards](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
+  * [Windows GPU Support](https://learn.microsoft.com/en-us/azure/aks/use-windows-gpu) on AKS is now available in preview.
+  * [Trusted Launch Support](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) on AKS is now available in preview.
+
+* Behavioral change
+  * Added new available type for `outbound-type`: `none`.
```
Comment (wangyira) [inline CHANGELOG.md:24]: What's outbound-type specifically used for? So users can find it easier

File: CHANGELOG.md (hunk 3 above)
Comment (alvinli222) [reply CHANGELOG.md:38]: good call out let me add

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,30 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-03-17
+
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).
+
+### Announcements
```
Comment (haitch) [inline CHANGELOG.md:7]: we need to announce deprecation of [List OrchestratorProfile API](https://learn.microsoft.com/en-us/rest/api/aks/container-services/list-orchestrators?view=rest-aks-2019-08-01&tabs=HTTP), it is fully replaced by [List KubernetesVersions API](https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/list-kubernetes-versions?view=rest-aks-2024-01-01&viewFallbackFrom=rest-aks-2023-11-01&tabs=HTTP)

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,48 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes is cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Changes to kube-reserved memory reservations are now in effect in AKS 1.29. The optimized reservation logic reduces kube-reserved memory by up to 20% depending on the node configuration. For existing 1.29 node pools created prior to 2/26, please perform a node pool update or recreate to see these changes. [Learn more.](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
```
Comment (allyford) [inline CHANGELOG.md:10]:
```suggestion
* Changes to kube-reserved memory reservations are now in effect in AKS 1.29. The optimized reservation logic reduces kube-reserved memory by up to 20% depending on the node configuration. For existing 1.29 node pools created prior to 2/26, please perform a node pool update or recreate to see these changes. [Learn more.](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
* On 15 March 2027, Windows Server 2022 will be retired when Kubernetes 1.34 reaches the end of platform support. You won't be able to create new Windows Server 2022 node pools on Kubernetes 1.35 and above. We encourage you to make the switch before 15 March 2027 to gain the richer benefits of [Windows Server 2025](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/introducing-windows-server-2025/ba-p/4026374) or Windows Server [Annual Channel](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248). These new Windows OS versions will be supported on AKS before Windows Server 2022 is retired. For more updates, see our [AKS public roadmap](https://github.com/Azure/AKS/projects/1).
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,48 @@
...
+
+* Behavioral change
+  * Added new available type for `outbound-type`: `none`.
+  * Workload Identity is now supported as a setting for static PVs on Managed Blob/File CSI drivers in 1.29.
+  * Starting with the `2024-03-01` api, `OSType` will reject unknown inputs.
+  * Added GPU support for new Nvidia ND SKU sizes.
+
+
+* Bug fixes
+  * AgentPool PUT requests with node init taints will no longer fail if the request does not change the field.
```
Comment (allyford) [inline CHANGELOG.md:31]: Remove this - its a fix to a feature that is about to release in preview.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,48 @@
...
+  * Fixed a bug where clusters running LTS could get a list of non-LTS versions to upgrade to.
+  * Fixed a bug with Application Gateway Ingress Controller where it is unable to fetch secret objects during cluster upgrade.
+
+
+* Component updates
+  * Upgraded azure-vnet from 1.4.39/1.4.43 to 1.4.52 and 1.5.11 to 1.5.23.
+  * Upgraded [AgentBaker](https://github.com/Azure/AgentBaker) version to v0.20240313.0
+  * Upgraded Linux [Network Policy Manager](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23) from 1.4.45.3 to 1.5.23
+  * AKS clusters running Cilium on 1.27 will be upgraded to Cilium 1.13 due to Cilium 1.12 EOL.
+
```
Comment (allyford) [inline CHANGELOG.md:43]: Remove extra spaces

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,48 @@
...
+* Bug fixes
+  * AgentPool PUT requests with node init taints will no longer fail if the request does not change the field.
+  * Fixed a bug with legacy clusters that blocked all new operations on these clusters.
+  * Fixed a bug where node taints may be overwritten on certain PUT requests.
+  * Fixed a bug where clusters running LTS could get a list of non-LTS versions to upgrade to.
+  * Fixed a bug with Application Gateway Ingress Controller where it is unable to fetch secret objects during cluster upgrade.
+
+
+* Component updates
+  * Upgraded azure-vnet from 1.4.39/1.4.43 to 1.4.52 and 1.5.11 to 1.5.23.
```
Comment (allyford) [inline CHANGELOG.md:39]: Any link?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,48 @@
...
+  * Added GPU support for new Nvidia ND SKU sizes.
+
+
+* Bug fixes
+  * AgentPool PUT requests with node init taints will no longer fail if the request does not change the field.
+  * Fixed a bug with legacy clusters that blocked all new operations on these clusters.
+  * Fixed a bug where node taints may be overwritten on certain PUT requests.
+  * Fixed a bug where clusters running LTS could get a list of non-LTS versions to upgrade to.
+  * Fixed a bug with Application Gateway Ingress Controller where it is unable to fetch secret objects during cluster upgrade.
+
```
Comment (allyford) [inline CHANGELOG.md:36]: Remove extra space

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,48 @@
...
+  * [Deployment Safeguards](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
+  * [Windows GPU Support](https://learn.microsoft.com/en-us/azure/aks/use-windows-gpu) on AKS is now available in preview.
+  * [Trusted Launch Support](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) on AKS is now available in preview.
+
+* Behavioral change
+  * Added new available type for `outbound-type`: `none`.
+  * Workload Identity is now supported as a setting for static PVs on Managed Blob/File CSI drivers in 1.29.
+  * Starting with the `2024-03-01` api, `OSType` will reject unknown inputs.
+  * Added GPU support for new Nvidia ND SKU sizes.
+
```
Comment (allyford) [inline CHANGELOG.md:28]: Remove extra space

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,48 @@
...
+* Changes to kube-reserved memory reservations are now in effect in AKS 1.29. The optimized reservation logic reduces kube-reserved memory by up to 20% depending on the node configuration. For existing 1.29 node pools created prior to 2/26, please perform a node pool update or recreate to see these changes. [Learn more.](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
+
+### Release notes
+
+* Features
+  * Kubernetes 1.29 is [GA](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli).
+  * [5,000 Node Limit by Default](https://azure.microsoft.com/en-us/updates/generally-available-azure-kubernetes-service-aks-support-for-5k-node-limit-by-default-for-standard-tier-clusters/) is generally available in AKS. This limit is available for Standard tier and Premium tier clusters.
```
Comment (allyford) [inline CHANGELOG.md:16]:
```suggestion
  * [5,000 Node Limit by Default](https://azure.microsoft.com/en-us/updates/generally-available-azure-kubernetes-service-aks-support-for-5k-node-limit-by-default-for-standard-tier-clusters/) is generally available in AKS. This limit is available for Standard tier and Premium tier clusters.
  * [Gen 2 VMs](https://learn.microsoft.com/azure/aks/cluster-configuration) are now generally available for Windows on AKS. Azure [Generation 2 (Gen2) virtual machines (VMs)](https://learn.microsoft.com/azure/virtual-machines/generation-2) support key features not supported in generation 1 VMs (Gen1).
  * [Custom kubelet configuration](https://learn.microsoft.com/azure/aks/custom-node-configuration?tabs=windows-node-pools) is now generally available for Windows on AKS. To request additional kubelet parameters supported by Windows, create a feature request on [AKS Github Issues](https://github.com/Azure/AKS/issues).
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,48 @@
...
+
+* Preview features
+  * [Deployment Safeguards](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
```
Comment (allyford) [inline CHANGELOG.md:19]:
```suggestion
  * [Deployment Safeguards](https://learn.microsoft.com/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,48 @@
...
+  * [Windows GPU Support](https://learn.microsoft.com/en-us/azure/aks/use-windows-gpu) on AKS is now available in preview.
```
Comment (allyford) [inline CHANGELOG.md:20]:
```suggestion
  * [Windows GPU Support](https://learn.microsoft.com/azure/aks/use-windows-gpu) on AKS is now available in preview.
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,48 @@
...
+  * [Trusted Launch Support](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) on AKS is now available in preview.
```
Comment (allyford) [inline CHANGELOG.md:21]:
```suggestion
  * [Trusted Launch Support](https://learn.microsoft.com/azure/aks/use-trusted-launch) on AKS is now available in preview.
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,48 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes is cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Changes to kube-reserved memory reservations are now in effect in AKS 1.29. The optimized reservation logic reduces kube-reserved memory by up to 20% depending on the node configuration. For existing 1.29 node pools created prior to 2/26, please perform a node pool update or recreate to see these changes. [Learn more.](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
+
+### Release notes
+
+* Features
+  * Kubernetes 1.29 is [GA](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli).
```
Comment (allyford) [inline CHANGELOG.md:15]:
```suggestion
  * Kubernetes 1.29 is [GA](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli).
```

**Review by wangyira — APPROVED** [review]

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,51 @@
...
+  * [Custom kubelet configuration](https://learn.microsoft.com/azure/aks/custom-node-configuration?tabs=windows-node-pools) is now generally available for Windows on AKS. To request additional kubelet parameters supported by Windows, create a feature request on [AKS Github Issues](https://github.com/Azure/AKS/issues).
+
+* Preview features
+  * [Deployment Safeguards](https://learn.microsoft.com/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
+  * [Windows GPU Support](https://learn.microsoft.com/azure/aks/use-windows-gpu) on AKS is now available in preview.
+  * [Trusted Launch Support](https://learn.microsoft.com/azure/aks/use-trusted-launch) on AKS is now available in preview.
+
+* Behavioral change
+  * Workload Identity is now supported as a setting for static PVs on Managed Blob/File CSI drivers in 1.29.
```
Comment (shashankbarsin) [inline CHANGELOG.md:28]: documentation link? cc: @miwithro

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,5 +1,51 @@
...
+  * Added GPU support for new Nvidia ND SKU sizes.
+
+* Bug fixes
+  * Fixed a bug with legacy clusters that blocked all new operations on these clusters.
+  * Fixed a bug where node taints may be overwritten on certain PUT requests.
+  * Fixed a bug where clusters running LTS could get a list of non-LTS versions to upgrade to.
+  * Fixed a bug with Application Gateway Ingress Controller where it is unable to fetch secret objects during cluster upgrade.
+
+* Component updates
+  * Upgraded [AgentBaker](https://github.com/Azure/AgentBaker) version to v0.20240313.0
```
Comment (shashankbarsin) [inline CHANGELOG.md:39]: we can remove this line as we don't capture agentbaker version explicitly in notes. we instead capture the vhd notes for different OS, which I see is added below

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,5 +1,51 @@
...
+  * Upgraded [Azure CNI](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23) from 1.4.39/1.4.43 to 1.4.52 and 1.5.11 to 1.5.23.
```
Comment (shashankbarsin) [inline CHANGELOG.md:40]:
```suggestion
  * Upgraded Azure CNI from 1.4.39/1.4.43 to [1.4.52](https://github.com/Azure/azure-container-networking/releases/tag/v1.4.52) and 1.5.11 to [1.5.23](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23).
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,51 @@
...
+  * Upgraded Linux [Network Policy Manager](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23) from 1.4.45.3 to 1.5.23
```
Comment (shashankbarsin) [inline CHANGELOG.md:41]:
```suggestion
  * Upgraded Linux Network Policy Manager from 1.4.45.3 to [1.5.23](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23)
```

**Review by haitch — APPROVED** [review]

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,5 +1,51 @@
...
+  * AKS clusters running Cilium on 1.27 will be upgraded to Cilium 1.13 due to Cilium 1.12 EOL.
```
Comment (shashankbarsin) [inline CHANGELOG.md:42]:
```suggestion
  * AKS clusters with Kubernetes version 1.27 and running Cilium will be upgraded to Cilium 1.13 due to Cilium 1.12 EOL.
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,5 +1,51 @@
...
+
+* Preview features
+  * [Deployment Safeguards](https://learn.microsoft.com/azure/aks/deployment-safeguards) on AKS is now available in preview with two levels of configuration: `Warning` and `Enforcement`
+  * [Windows GPU Support](https://learn.microsoft.com/azure/aks/use-windows-gpu) on AKS is now available in preview.
+  * [Trusted Launch Support](https://learn.microsoft.com/azure/aks/use-trusted-launch) on AKS is now available in preview.
+
+* Behavioral change
+  * Workload Identity is now supported as a setting for static PVs on Managed Blob/File CSI drivers in 1.29.
+  * Starting with the `2024-03-01` api, `OSType` will reject unknown inputs.
+  * Added GPU support for new Nvidia ND SKU sizes.
```
Comment (shashankbarsin) [inline CHANGELOG.md:30]: do we have a list of the new SKUs?

**Review by seguler — APPROVED** [review]

**Review by seguler — APPROVED** [review]

**Pull request merged**